src/ObjCRuntime/Runtime.cs

```diff
@@ -1225,6 +1249,37 @@ static void MissingCtor (IntPtr ptr, IntPtr klass, Type type, MissingCtorResolut
 			}
 
 			msg.Append (").");
+			if (sel != IntPtr.Zero || method_handle.Value != IntPtr.Zero) {
+				msg.AppendLine ();
+				msg.AppendLine ("Additional information:");
+				if (sel != IntPtr.Zero)
+					msg.Append ("\tSelector: ").Append (Selector.GetName (sel)).AppendLine ();
+				if (method_handle.Value != IntPtr.Zero) {
+					try {
+						var method = MethodBase.GetMethodFromHandle (method_handle);
+						msg.Append ($"\tMethod: ");
+						if (method is not null) {
+							// there's no good built-in function to format a MethodInfo :/
+							msg.Append (method.DeclaringType?.FullName ?? string.Empty);
+							msg.Append (".");
+							msg.Append (method.Name);
+							msg.Append ("(");
+							var parameters = method.GetParameters ();
+							for (var i = 0; i < parameters.Length; i++) {
+								if (i > 0)
+									msg.Append (", ");
+								msg.Append (parameters [i].ParameterType.FullName);
+							}
```

Comment:
No need to implement our own AppendJoin: https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.appendjoin?view=net-8.0#system-text-stringbuilder-appendjoin(system-string-system-string())

Reply:
TIL!